python/freetensor/core/staging.py

```diff
@@ -246,10 +251,18 @@ def load_attr(self, obj, attr: str):
         try:
             return getattr(obj, attr)
         except AttributeError:
-            loaded = self.custom_attr(obj, attr)
-            if loaded:
-                return loaded
-            raise
+            try:
+                # Have to use AttributeError again, since a custom attribute might have
+                # a None value
+                result = self.custom_attr(obj, attr)
+                successful = True
+            except AttributeError:
+                successful = False
+
+            if successful:
+                return result
+            else:
+                raise
```

Comment:
We are raising the outer `AttributeError` here. Why is this decision? What about raising the inner `AttributeError` directly from Line 260?